configure

```diff
@@ -2172,7 +2173,7 @@ if test "$have_gcc" = yes ; then
 		case $_host_os in
 		# newlib-based system include files suppress non-C89 function
 		# declarations under __STRICT_ANSI__, undefine it
-		3ds | amigaos* | android | androidsdl | dreamcast | ds | gamecube | mingw* | mint* | n64 | psp | ps3 | psp2 | switch | wii )
+		3ds | amigaos* | android | androidsdl | dreamcast | ds | gamecube | mint* | n64 | psp | ps3 | psp2 | switch | wii )
```

Comment:
This change doesn't seem related to the other ones and I am not sure at all about it.

Reply:
I had a warning on every file when `__STRICT_ANSI__` was undefined (with gcc 11 and mingw 9).

Reply:
@lotharsm do you see any trouble with this?
I am not sure how it would work with older versions of mingw?
Is it newlib based after all?

Reply:
@lephilousophe I don't see issues with it, and I'm curious right now why it is even in there. Looks like it _might_ only affect really ancient versions or specific builds. It might even be totally irrelevant on current versions of the C++ standard.

The only references regarding `newlib` I could find are in very old forums posts from back in 2006. MSYS2 only refers to it in conjunction with embedded ARM stuff (which is expected since newlib was created for embedded devices) and some RISCV packages [1]. I don't think newlib is used on current and "regular" MinGW instances.

I can confirm that at least on MSYS2/Mingw-w64, these changes work and don't harm anything.

To me, it looks like the `mingw*` entry here is a leftover from quite some time back ago.

[1] https://packages.msys2.org/search?q=newlib

Reply:
Hehe, as you don't know why I followed the inclusion of this in configure script.
It was there from the beginning (472ca9c68976a99802ed44dfe7399a2b11b248a5, 2004!) of mingw in configure. No mention about newlib at that time but maybe the mingw from these ancient times hide the functions if -ansi was used.
So we can assume it's not the case anymore.

Reply:
This causes a compilation failure with MinGW32 (using GCC 9.2.0 and mingwrt 5.4.2):

```
backends/fs/stdiostream.cpp: In static member function 'static StdioStream* StdioStream::makeFromPath(const Common::String&, bool)':
backends/fs/stdiostream.cpp:121:17: error: '_wfopen' was not declared in this scope; did you mean '_lopen'?
  121 |  FILE *handle = _wfopen(wPath, writeMode ? L"wb" : L"rb");
      |                 ^~~~~~~
      |                 _lopen
```

The Switch port also warns when `__STRICT_ANSI__` is undefined, but fails when removed from this case statement, so I'd suggest taking a slightly different approach.

Reply:
The `__STRICT_ANSI__` issue is new with GCC11. See https://github.com/actions/virtual-environments/issues/3376.

I'll revert it then. Maybe later we'll revisit this and make the undefine conditional.